graphql/fetchUserProfileCard.ts

```diff
@@ -1,38 +1,50 @@
 import { gql } from "@apollo/client";
-import React from "react";
+
 export type Users = {
   id: string;
   name: string;
   link: string;
   profile_image: string;
-  badges_earned: number;
   created_at: string;
   current_badge: number;
   coding_badge: CodingBadge;
+  user_coding_badges_aggregate: BadgesEarnedCount;
 };
 
 export type CodingBadge = {
   title: string;
   image: string;
 };
 
+export type BadgesEarnedCount = {
```

Comment:
cc: @LuckyHariharan 

I learned a new typescript trick recently - when we have nested fields, we don't _have_ to declare a new type for each nested field. Take a look at this:

```
export type Users = {
  id: string;
  name: string;
  link: string;
  profile_image: string;
  created_at: string;
  current_badge: number;
  coding_badge: CodingBadge;
  user_coding_badges_aggregate: {
    aggregate: {
      count: number;
    };
  };
};
```
you don't need the BadgesEarnedCount and BadgesEarnedNested types any more

In some cases it does make sense to have separate nested types like coding badges

Reply:
Great idea.  I had a couple of those on this PR and I changed them all!